src/System.Linq/tests/OrderByTests.cs

```diff
@@ -441,5 +443,109 @@ public void TakeOne(IEnumerable<int> source)
             }
             Assert.Equal(1, count);
         }
+
+        [Fact]
+        public void CultureOrderBy()
+        {
+            CultureInfo current = CultureInfo.CurrentCulture;
+            try
+            {
+                string[] source = new[] { "Apple0", "�ble0", "Apple1", "�ble1", "Apple2", "�ble2" };
+
+                CultureInfo dk = new CultureInfo("da-DK");
+                CultureInfo au = new CultureInfo("en-AU");
+
+                StringComparer comparerDk = StringComparer.Create(dk, false);
+                StringComparer comparerAu = StringComparer.Create(au, false);
+
+                // we don't provide a defined sorted result set because the Windows culture sorting
+                // provides a different result set to the Linux culture sorting. But as we're really just
+                // concerned that OrderBy default string ordering matches current culture then this
+                // should be sufficient
+                string[] resultDK = source.ToArray();
+                Array.Sort(resultDK, comparerDk);
+                string[] resultAU = source.ToArray();
+                Array.Sort(resultAU, comparerAu);
+
+                string[] check;
+
+                CultureInfo.CurrentCulture = dk;
+                check = source.OrderBy(x => x).ToArray();
+                Assert.Equal(check, resultDK, StringComparer.Ordinal);
```

Comment:
The first parameter is expected, second is actual, so these should be reversed. (Same in other such cases below.)